### PR TITLE
Spatial operations

### PR DESCRIPTION
**Pull request by ThadT**
Spatial (geometry) operations sample for all platforms. Uses two overlapping polygon graphics to illustrate `GeometryEngine` spatial operations like `Difference`, `Intersection`, etc.

Please review when you have a chance, @nCastle1 
Note that the PR is against v.next. This could go into master if needed (would have to address some conflicts, etc.).

### CODE REVIEW

**Comment by nCastle1** [comment]
I'd be happy to review this! Let's hold off on merging (regardless of whether changes are needed) due to the significant restructure of the repo that is waiting in PR. I can work with you to move the samples to the new structure once that branch has been merged.

File: src/WPF/ArcGISRuntime.WPF.Samples/Samples/GeometryEngine/SpatialOperations/SpatialOperations.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.UI;
+using System.Collections.Generic;
+using System.Windows;
+using Win = System.Windows.Media;
```
[the remark this answers is not in the document]
Comment (ThadT) [reply src/WPF/ArcGISRuntime.WPF.Samples/Samples/GeometryEngine/SpatialOperations/SpatialOperations.xaml.cs]: Thanks, sounds like we might need to add a standard for aliasing usings too!

File: src/iOS/Xamarin.iOS/Samples/GeometryEngine/SpatialOperations/SpatialOperations.cs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        private Graphic _graphicTwo;
+
+        // Graphic to display the spatial operation result polygon.
+        private Graphic _resultGraphic;
+
+        // Stack view to contain the spatial operation UI.
+        private UIStackView _operationToolsView = new UIStackView();
+
+        // Store the height of each set of controls (may vary on different devices).
+        private nfloat _mapViewHeight;
```
[the remark this answers is not in the document]
Comment (ThadT) [reply src/iOS/Xamarin.iOS/Samples/GeometryEngine/SpatialOperations/SpatialOperations.cs]: Good catch

**Comment by ThadT** [comment]
Cool, thanks @nCastle1 
Sorry ... I merged this into v.next without thinking. Let me know when we're ready to move this into the new format.

**Comment by nCastle1** [comment]
No worries, I'll let you know when things are ready to merge. In the meantime, let's hold off on deleting the branch.

**Pull request merged**